Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnUpdate.hs

```diff
@@ -52,7 +55,9 @@ data OnUpdateBuildReq
       }
   | RideCompletedBuildReq
       { ride :: DRide.Ride,
-        fareParams :: Fare.FareParameters
+        fareParams :: Fare.FareParameters,
+        startLocation :: Maybe DLocation.Location,
```

Comment:
Why you are passing start location in `complete` request?

Reply:
from Driver side when the ride will end we will be passing start and end location to update on the customer side using onupdate as customer's ride will also be ended at the same time so using this we are updating customer side. We are not passing lat longs here, directly the location so that we don't have to call the api for fetching place name again

Reply:
the use case was updating the ride list so that if customer changes pickup or drop location in this rides tab he should be shown only the actual pickup or drop locations so we need it only at the end of the ride

Reply:
> the use case was updating the ride list so that if customer changes pickup or drop location in this rides tab he should be shown only the actual pickup or drop locations so we need it only at the end of the ride

I got this much, but as i said `startLocation` is obviously known at `Start` stage, user should not be able to change start location during the ride, it doesn't make any sense. So why it is sent here and not in `StartRideReq`?

Reply:
rider will not be able to change start location after ride.

Reply:
> rider will not be able to change start location after ride.

That's not what i'm asking. I'm asking why you are passing and setting startLocation with `EndRideReq`, when by any imaginable logic it should be passed and set with `StartRideReq`

Reply:
@NikithShetty can you check if this is as per the requirement?

Reply:
Sure @DonDokidon we can change this to startRide .